RFS/src/main/java/com/rfs/transformers/Transformer_ES_6_8_to_OS_2_11.java

```diff
@@ -1,38 +1,47 @@
 package com.rfs.transformers;
 
+import java.util.List;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.rfs.models.GlobalMetadata;
+import com.rfs.models.IndexMetadata;
+import com.rfs.version_os_2_11.GlobalMetadataData_OS_2_11;
+
+import org.opensearch.migrations.transformation.TransformationRule;
+import org.opensearch.migrations.transformation.entity.Index;
+import org.opensearch.migrations.transformation.rules.IndexMappingTypeRemoval;
 
 public class Transformer_ES_6_8_to_OS_2_11 implements Transformer {
     private static final Logger logger = LogManager.getLogger(Transformer_ES_6_8_to_OS_2_11.class);
     private static final ObjectMapper mapper = new ObjectMapper();
+
+    private final List<TransformationRule<Index>> indexTransformations = List.of(new IndexMappingTypeRemoval());
     private int awarenessAttributeDimensionality;
 
     public Transformer_ES_6_8_to_OS_2_11(int awarenessAttributeDimensionality) {
         this.awarenessAttributeDimensionality = awarenessAttributeDimensionality;
     }
 
     @Override
-    public ObjectNode transformGlobalMetadata(ObjectNode root) {
+    public GlobalMetadata transformGlobalMetadata(GlobalMetadata globalData) {
+        var root = globalData.toObjectNode();
         ObjectNode newRoot = mapper.createObjectNode();
 
         // Transform the original "templates", but put them into the legacy "templates" bucket on the target
-        if (root.get("templates") != null) {
-            ObjectNode templatesRoot = (ObjectNode) root.get("templates").deepCopy();
-            templatesRoot.fieldNames().forEachRemaining(templateName -> {
-                ObjectNode template = (ObjectNode) templatesRoot.get(templateName);
-                logger.info("Transforming template: " + templateName);
-                logger.debug("Original template: " + template.toString());
-                TransformFunctions.removeIntermediateMappingsLevels(template);
-                TransformFunctions.removeIntermediateIndexSettingsLevel(template); // run before fixNumberOfReplicas
-                TransformFunctions.fixReplicasForDimensionality(template, awarenessAttributeDimensionality);
-                logger.debug("Transformed template: " + template.toString());
-                templatesRoot.set(templateName, template);
+        var originalTemplates = root.get("templates");
+        if (originalTemplates != null) {
+            var templates = mapper.createObjectNode();
+            originalTemplates.fieldNames().forEachRemaining(templateName -> {
+                var templateCopy = (ObjectNode) originalTemplates.get(templateName).deepCopy();
+                var indexTemplate = (Index) () -> templateCopy;
+                transformIndex(indexTemplate);
```

Comment:
Oh, I think I finally see now what's going on.  We're applying a list of transformations for index JSON to all the of the template JSON as well?  This is surprising to me.  Why not have separate transformation lists, which may share similar sets of transforms?

Reply:
I've created a separate list and changed how the private function works so it passes if it is a concrete or abstract index.